Analytics-CSharp/Segment/Analytics/Analytics.cs

```diff
@@ -116,9 +119,9 @@ public void Process(RawEvent incomingEvent)
         /// </summary>
         public void Reset()
         {
+            // update cache and persist storage
             string newAnonymousId = Guid.NewGuid().ToString();
             _userInfo = new UserInfo(newAnonymousId, null, null);
-
             AnalyticsScope.Launch(AnalyticsDispatcher, async () =>
             {
                 await Store.Dispatch<UserInfo.ResetAction, UserInfo>(new UserInfo.ResetAction(newAnonymousId));
```

Comment:
Oh, I understand now what you meant with the `cache`.
Is there any reason we need to keep around the `userInfo` in the state? Won't that one still have the same issue though? e.g. at some point the cache vs store userInfo values might deviate due to one being updated sync and the other one async.

Reply:
Not for this scenario for sure, but if potentially a plugin tries to get the userInfo from the state / set the userInfo it might see different values.

I think we should try to lock down plugins to only read/update the values from the event itself, that way data flows only one way + there's only one source of truth (the cache) + data doesn't mutate within async operations.

Reply:
yeah. we actually don't have to do state management on user info, since all we need is to write to the cache and then persist the data in the background. in that case, we always read from the cache, which guaranteed to be the latest copy. the only edge case is the background thread not yet finishing persist the most recent copy, but app gets terminated. then on app start up, we're not reading the most recent user info. 

but if we have state management, then there are more edge cases we need to worry about. anytime there is a new subscriber to user info, there is a chance of inconsistent user info. the reason I still keep using it is:
1. more aligned logic across other libraries
2. centralize all the storage related code in one place

completely agree we should always read from the in-memory copy instead of state management. but I guess all we can do is making a note inline with the code?